pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,106 @@
+package telemetry
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"os"
+	"time"
+
+	"net/http"
+	"net/url"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+// eventType
+const eventType = "library-os-ng"
+
+// actionName
+const actionName = "Piper Library OS"
+
+// LibraryRepository that is passed into with -ldflags
+var LibraryRepository string
+
+// SiteID ...
+var SiteID string
+
+var disabled bool
+var client piperhttp.Sender
+
+// Initialize sets up the base telemetry data and is called in generated part of the steps
+func Initialize(telemetryDisabled bool, stepName string) {
+	disabled = telemetryDisabled
+
+	// skip if telemetry is dieabled
+	if disabled {
+		return
+	}
+
+	if client == nil {
+		client = &piperhttp.Client{}
+	}
+
+	client.SetOptions(piperhttp.ClientOptions{Timeout: time.Second * 5})
+
+	if len(LibraryRepository) == 0 {
+		LibraryRepository = "https://github.com/n/a"
+	}
+
+	if len(SiteID) == 0 {
+		SiteID = "827e8025-1e21-ae84-c3a3-3f62b70b0130"
+	}
+
+	baseData = BaseData{
+		URL:             LibraryRepository,
+		ActionName:      actionName,
+		EventType:       eventType,
+		StepName:        stepName,
+		SiteID:          SiteID,
+		PipelineURLHash: getPipelineURLHash(), // http://server:port/jenkins/job/foo/
+		BuildURLHash:    getBuildURLHash(),    // http://server:port/jenkins/job/foo/15/
+	}
+	//ToDo: register Logrus Hook
+
+}
+
+func getPipelineURLHash() string {
+	return toSha1OrNA(os.Getenv("JOB_URL"))
+}
+
+func getBuildURLHash() string {
+	return toSha1OrNA(os.Getenv("BUILD_URL"))
+}
+
+func toSha1OrNA(input string) string {
+	if len(input) == 0 {
+		return "n/a"
+	}
+	return fmt.Sprintf("%x", sha1.Sum([]byte(input)))
+}
+
+// SWA baseURL
+const baseURL = "https://webanalytics.cfapps.eu10.hana.ondemand.com"
+
+// SWA endpoint
+const endpoint = "/tracker/log"
+
+// SendTelemetry ...
+func SendTelemetry(customData *CustomData) {
+	data := Data{
+		BaseData:     baseData,
+		BaseMetaData: baseMetaData,
+		CustomData:   *customData,
+	}
+
+	// skip if telemetry is dieabled
+	if disabled {
```

Comment:
What about emitting a log message?

Reply:
From my point of view logging during `Initialize` would be sufficient, what do you think.